components/GoalsForm/GoalsForm.js

```diff
@@ -0,0 +1,108 @@
+import React, { useState } from "react";
+import styled from "styled-components";
+import { StyledHint } from "@/styles";
+
+const ModalBackround = styled.div`
+  position: fixed;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: rgba(0, 0, 0, 0.5);
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const ModalContainer = styled.div`
+  background: white;
+  padding: 20px;
+  border-radius: 10px;
+  box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+`;
+
+const GoalSubmitButton = styled.button`
+  position: fixed;
+  bottom: 5rem;
+  right: 2rem;
+  background-color: purple;
+  color: white;
+  border: none;
+  border-radius: 50%;
+  width: 3rem;
+  height: 3rem;
+  font-size: 24px;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+export default function GoalsForm({ onAddGoal }) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  function handleSubmit(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const data = Object.fromEntries(formData);
+
+    onAddGoal({
+      ...data,
+      savedAmount: parseInt(data.savedAmount),
+      goalAmount: parseInt(data.goalAmount),
+    });
+    setIsModalOpen(false);
+
+    console.log("Modal Open", isModalOpen);
+    console.log("Event", event);
+    console.log("data", data);
+    console.log("Formdata", formData);

```

Comment:
I like that we are console logging. This helps with development, but we don't want these logs appearing in production. Please remove them.